### PR TITLE
show cumulative data at turn instructions cell

### PR DESCRIPTION
**Pull request by nnngrach**
[issue](https://github.com/osmandapp/OsmAnd-iOS/issues/2981#issuecomment-2147143395)

### Show cumulative data on labels

Before
before Screenshot 2024-06-04 at 15 37 32

After
after Screenshot 2024-06-04 at 15 58 28

---

### Return disappeared info row

Before
Screenshot 2024-06-04 at 16 12 00

After
Screenshot 2024-06-04 at 16 37 21

---

### Fixed turn lane text size and color for roundabout roads

before / after
Screenshot 2024-06-04 at 16 45 09 Screenshot 2024-06-04 at 17 08 05

Screenshot 2024-06-04 at 16 46 10 Screenshot 2024-06-04 at 17 08 08

---

Hide extra top separator

Before / after
Screenshot 2024-06-04 at 21 34 07 Screenshot 2024-06-04 at 21 32 43

---

Use iconColorDefault for stroke, lanes widget.
Try to increase height of arrows in lanes, height = 30 pt
Padding from the screen side to separator 76 pt
Turn distance should respect Distance during navigation option

Before / after
Screenshot 2024-06-05 at 18 29 02 Screenshot 2024-06-05 at 18 23 26

Screenshot 2024-06-05 at 18 29 16 Screenshot 2024-06-05 at 18 23 45

### CODE REVIEW

File: Sources/Controllers/Cells/Xibs/RouteInfoListItemCell.xib [hunk 1]
```diff
@@ -40,16 +40,16 @@
                                 <rect key="frame" x="56" y="0.0" width="239" height="72"/>
                                 <subviews>
                                     <stackView opaque="NO" contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="iGS-Y6-iIS">
-                                        <rect key="frame" x="0.0" y="0.0" width="239" height="16"/>
+                                        <rect key="frame" x="0.0" y="0.0" width="239" height="15.666666666666666"/>
                                         <subviews>
                                             <label opaque="NO" userInteractionEnabled="NO" contentMode="left" verticalHuggingPriority="251" text="922 m" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="OQB-OA-v6f">
-                                                <rect key="frame" x="0.0" y="0.0" width="146.66666666666666" height="16"/>
+                                                <rect key="frame" x="0.0" y="0.0" width="146.66666666666666" height="15.666666666666666"/>
```
Comment (tigrim) [inline Sources/Controllers/Cells/Xibs/RouteInfoListItemCell.xib:46]: ?

File: Sources/Controllers/Cells/Xibs/RouteInfoListItemCell.xib [hunk 2]
```diff
@@ -58,20 +58,23 @@
                                         <color key="backgroundColor" white="0.0" alpha="0.0" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
                                     </stackView>
                                     <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="You have arrived." textAlignment="natural" lineBreakMode="tailTruncation" numberOfLines="0" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="gcJ-62-qo7">
-                                        <rect key="frame" x="0.0" y="20" width="239" height="18"/>
+                                        <rect key="frame" x="0.0" y="19.666666666666664" width="239" height="10.333333333333336"/>
```
Comment (tigrim) [inline Sources/Controllers/Cells/Xibs/RouteInfoListItemCell.xib:61]: ?

File: Sources/Controllers/Cells/RouteInfoListItemCell.swift [hunk 3]
```diff
@@ -56,16 +60,24 @@ final class RouteInfoListItemCell: UITableViewCell {
...
         if traitCollection.hasDifferentColorAppearance(comparedTo: previousTraitCollection) {
             refreshLeftIcon()
+            bottomImageBorderView.layer.borderColor = UIColor.iconColorDefault.cgColor
         }
     }
     
     private func refreshLeftIcon() {
         guard let leftTurnIconDrawable else { return }
-        leftTurnIconDrawable.clr = (traitCollection.userInterfaceStyle == .dark) ? .white : .black
+        let color = (traitCollection.userInterfaceStyle == .dark) ? UIColor.widgetValue.dark : UIColor.widgetValue.light
```
Comment (tigrim) [inline Sources/Controllers/Cells/RouteInfoListItemCell.swift:78]: Can you use this code?` let color =  UIColor.widgetValue.dark`

File: Sources/Controllers/Cells/OASegmentTableViewCell.m [hunk 4]
```diff
@@ -20,4 +20,22 @@ - (void)setSelected:(BOOL)selected animated:(BOOL)animated
     [super setSelected:selected animated:animated];
 }
 
+- (void)layoutSubviews
+{
+    [super layoutSubviews];
+    
+    for (UIView *view in self.subviews)
+    {
+        if ([NSStringFromClass(view.class) hasSuffix:@"CellSeparatorView"])
```
Comment (tigrim) [inline Sources/Controllers/Cells/OASegmentTableViewCell.m:29]: What does 'CellSeparatorView' mean? Please refrain from using private APIs. There's no guarantee that in future versions of iOS, Apple will use the same name

File: Sources/Controllers/Cells/RouteInfoListItemCell.swift (hunk 3 above)
Comment (nnngrach) [reply Sources/Controllers/Cells/RouteInfoListItemCell.swift:78]: I can. But clor will be always white. 
Screenshot 2024-06-07 at 17 14 48

**Comment by nnngrach** [comment]
found and fixed bug with expanding cell
https://github.com/osmandapp/OsmAnd-iOS/assets/35684515/f595f3ab-6bab-4493-b2a9-32710a19cb8f

**Comment by nnngrach** [comment]
video:
https://github.com/osmandapp/OsmAnd-iOS/assets/35684515/e66aaa5b-a9e4-451e-8bc7-04e11df54e0d

Screenshot 2024-06-10 at 22 12 16 Screenshot 2024-06-10 at 22 12 24
Screenshot 2024-06-10 at 22 12 39 Screenshot 2024-06-10 at 22 12 48

**Comment by nnngrach** [comment]
Screenshot 2024-06-11 at 13 41 45
Screenshot 2024-06-11 at 13 57 22

**Review by tigrim — APPROVED** [review]

**Pull request merged**